src/MapfileStyleParser.ts

```diff
@@ -705,17 +761,40 @@ export class MapfileStyleParser implements StyleParser {
   readStyle(mapfileString: string): Promise<Style> {
     return new Promise<Style>((resolve, reject) => {
       try {
-        const mapfileObject: object = parse(mapfileString);
-        // console.log(JSON.stringify(mapfileObject, null, 2));
-        const geoStylerStyle: Style = this.mapfileObjectToGeoStylerStyle(mapfileObject);
-        // console.log(JSON.stringify(geoStylerStyle, null, 2));
+        const mapfileObject: any = parse(mapfileString);
+        const mapfileLayers = mapfileObject?.map?.layer || [];
+        if (mapfileLayers.length > 1) {
+          throw new Error('Can not read multiple LAYER in one file. Use method readMultiStyle instead.');
+        }
+        const geoStylerStyle: Style = this.mapfileLayerToGeoStylerStyle(mapfileLayers[0]);
         resolve(geoStylerStyle);
       } catch (error) {
         reject(error);
       }
     });
   }
 
+  /**
+   * Same as readStyle but read mutliple LAYER in a mapfile.
+   *
+   * @param  {string} mapfileString A Mapfile as a string.
+   * @return {Promise} The Promise resolving with an array of GeoStyler-Style Style
+   */
+  readMultiStyles(mapfileString: string): Promise<Style[]> {
```

Comment:
Do we even want to provide this here? Maybe the users should handle this since they probably need to do some preprocessing anyway.

Reply:
I would like to keep it. It's usefull for the test (not creating 3 file to have full coverage on scales).
And having multiple layers in a mapfile is a really standard use case.